legate/core/runtime.py

```diff
@@ -327,6 +328,11 @@ def allocate_field(self) -> tuple[Region, int]:
     def free_field(
         self, region: Region, field_id: int, ordered: bool = False
```

Comment:
It looks like at this point this operation is always ordered (either because it's invoked after a consensus match, or because it's happening on a single-rank execution, in which case there is no other process to diverge from in terms of Python garbage collection). Otherwise it wouldn't be safe to emit a discard operation unconditionally. I suggest we remove the "ordered" argument from this method, and leave a comment that it's always implicitly ordered.

Reply:
Unfortunately, we do need an unordered path for discard operations even on a single rank. The key role that unordered paths play is to avoid CFFI calls in Python destructors, as doing otherwise can cause hang.

Reply:
https://github.com/nv-legate/legate.core/pull/630/commits/cc354e56046fbd694a1e5e4a93a81af6b4bbed53